### ISSUE TITLE
[storage] encryption

### ISSUE THREAD

**Issue by cchance27** [louis030195/screen-pipe#18]
I'm not seeing any mention of in-memory or at-rest encryption, given that was the MAJOR red flag from MS's recall... i'd hope that would be early on the list of things to be implemented.

If your PC is ever hacked you don't want to have a single target file/files that literally gets them access to your entire visual and audio history.

**Comment by Arche151** [comment]
I agree, implementing encryption of the files should be priority #1.  Before that, I cannot use screen-pipe, unfortunately.

**Comment by louis030195** [comment]
thanks for the feedback, should be easy to implement :) will do, (or feel free to send PR)

**Comment by louis030195** [comment]
5 ways to encrypt data, which one would you prefer?

• Whole-database encryption
• File-level encryption
• In-memory encryption, writing encrypted data to disk
• Transparent disk encryption (e.g., LUKS on Linux, FileVault on macOS)
• Hybrid approach: encrypt sensitive fields, leave searchable data unencrypted

**Comment by louis030195** [comment]
/bounty 100

$100 for encryption of mp4 at rest (not the database yet)

definition of done:
- mp4 are encrypted on disk as soon as written
- when the user try to access it programmatically or through UI can unlock it (somehow automatically)

**Comment by higumachan** [comment]
@louis030195 

Would the encryption for MP4s involve encrypting the entire MP4 file? 
Another option could be to use chunk-based encryption, for example the Common Encryption Scheme (CENC).

Though, I do think CENC could be quite painful to implement.

**Comment by louis030195** [comment]
duplicate of #466
